sql/core/src/test/scala/org/apache/spark/sql/sources/StagingInsertSuite.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.sources
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.{QueryTest, Row}
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.util.Utils
+
+class StagingInsertSuite extends QueryTest with SharedSparkSession {
+  import testImplicits._
+
+  val stagingDir = Utils.createTempDir()
```

Comment:
be nice if this could be subclassed so we could add tests which tried this on, say abfs or gcs

Reply:
> be nice if this could be subclassed so we could add tests which tried this on, say abfs or gcs

You mean, use abfsm gcs as staging dir?

Reply:
i mean write tests which can be subclassed and then retargeted at a remote store. a lot of the spark relations test assume local fs everywhere, for example, and i had to copy and paste for cloud storage testing

Reply:
> i mean write tests which can be subclassed and then retargeted at a remote store. a lot of the spark relations test assume local fs everywhere, for example, and i had to copy and paste for cloud storage testing

Got it, let me try

Reply:
@steveloughran How about current? Is it what you want?